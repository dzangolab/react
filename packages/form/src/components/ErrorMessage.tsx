import { FC } from "react";

interface IErrorMessage {
  message: string;
}

export const ErrorMessage: FC<IErrorMessage> = ({ message }) => {
  return <span role="alert">{message}</span>;
};
