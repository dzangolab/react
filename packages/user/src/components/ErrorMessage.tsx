import React from "react";

interface Properties {
  touched?: boolean;
  error?: string;
}

const ErrorMessage = ({ touched, error }: Properties) => {
  return error && touched ? <p role="alert">{error}</p> : null;
};

export default ErrorMessage;
