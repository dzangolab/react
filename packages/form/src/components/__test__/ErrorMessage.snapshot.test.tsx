import React from "react";
import renderer, { create } from "react-test-renderer";
import { expect, test } from "vitest";

import { ErrorMessage } from "../ErrorMessage";

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();

  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);

  return result as renderer.ReactTestRendererJSON;
}

test("Component matches snapshot", () => {
  const component = create(<ErrorMessage message="Error message" />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
