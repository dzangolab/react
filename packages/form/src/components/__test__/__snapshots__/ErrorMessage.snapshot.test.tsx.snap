// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component matches snapshot 1`] = `
<span
  role="alert"
>
  Error message
</span>
`;
